4/Step 2/Shipper.ts

```diff
@@ -0,0 +1,30 @@
+import { AirEastShipper } from "./AirEastShipper";
+import { ChicagoSprintShipper } from "./ChicagoSprintShipper";
+import { PacificParcelShipper } from "./PacificParcelShipper";
+
+export abstract class Shipper {
+    public static shipper: Shipper;
+    public shipmentCost: number;
+
+    public static getInstance(fromZipCode: string): Shipper {
+        if (!Shipper.shipper) {
+            switch (fromZipCode[0]) {
+                case '4':
+                case '5':
+                case '6':
+                    return new ChicagoSprintShipper();
+                case '7':
+                case '8':
+                case '9':
```

Comment:
This is a very bad practice. You should always think about code readabilty when writing. Please refactor this and try different approach